code/go/0chain.net/blobbercore/reference/referencepath.go

```diff
@@ -78,6 +78,30 @@ func GetReferencePathFromPaths(ctx context.Context, allocationID string, paths [
 	return &refs[0], nil
 }
 
+func PathExists(ctx context.Context, allocationID string, path string) (bool, error) {
+	path = filepath.Clean(path)
+
+	if path == "." || path == "/" {
+		return false, nil
+	}
+
+	var refs []Ref
+	db := datastore.GetStore().GetTransaction(ctx)
+
+	ref := Ref{Path: path, AllocationID: allocationID}
+	db = db.Where(ref).Or("path LIKE ? AND allocation_id = ?", (path + "/%"), allocationID)
```

Comment:
do you think `db = db.Where(ref).Or("path = ? AND allocation_id = ?", path , allocationID)` is enough?